### PR TITLE
MRG, DOC: get docs to build on Windows

### PR DESCRIPTION
**Pull request by kalenkovich**
- set encoding of "commands.rst.new" to 'utf8',
- do not use memory_profiler on Windows,
- add graphviz as a requirement and the conda command to install it,
- add optipng as a requirement with ways to install it on Linux/MacOS/Windows

### CODE REVIEW

**Comment by larsoner** [comment]
@kalenkovich I realized what the problem might be. Have you downloaded the data for those examples yet? Sometimes it takes some minutes, and other times you actually have to accept some license terms -- so it's probably waiting for a `y` input that never comes.

For example:
```
 python -ui examples/preprocessing/plot_muscle_detection.py

License
-------
This tutorial dataset (EEG and MRI data) remains a property of the MEG Lab,
McConnell Brain Imaging Center, Montreal Neurological Institute,
McGill University, Canada. Its use and transfer outside the Brainstorm
tutorial, e.g. for research purposes, is prohibited without written consent
from the MEG Lab.

If you reference this dataset in your publications, please:

    1) acknowledge its authors: Elizabeth Bock, Esther Florin, Francois Tadel
       and Sylvain Baillet, and
    2) cite Brainstorm as indicated on the website:
       http://neuroimage.usc.edu/brainstorm

For questions, please contact Francois Tadel (francois.tadel@mcgill.ca).
Agree (y/[n])?
```
This will just hang. I can reproduce the hang on my system without that dataset with:
```
$ PATTERN=muscle make html_dev-pattern
```
I'll see if it's possible to get a fix into sphinx-gallery for this (it should probably raise an error instead of hanging). In the meantime, you can manually download the datasets, or use the private call `mne.datasets._download_all_example_data(verbose=True)`, which will take you through all necessary datasets, **but understand that you are agreeing to all terms of all datasets by doing so** (which usually just means agreeing to cite appropriate papers if you use the data).

But really building all docs locally is typically overkill, you should really just use `PATTERN=... make html_dev-pattern` to run just the example(s) of interest.

**Comment by kalenkovich** [comment]
No, I haven't downloaded the datasets. That must be it (can't really test right now due to space issues)! Thanks for figuring another one out. Trying to build docs triggered downloading of some of the datasets so I figured all of them got downloaded. I was obviously wrong.

I think the things you've helped me with so far should be enough to run `PATTERN=... make html_dev-pattern` on Windows which is what I was going for initially. So, I think this one is ready for review.

BTW, `PATTERN=... make html_dev-pattern` is

```shell
set PATTERN=...
make html_dev-pattern
```

Should I change this one too?

**Comment by larsoner** [comment]
> Should I change this one too?

It depends on what shell you're using in Windows (command prompt versus mingw or linux subtools, etc.). But feel free to add a note for windows users about this, since probably enough will use the builtin conda prompt.

**Comment by larsoner** [comment]
FYI https://github.com/sphinx-gallery/sphinx-gallery/pull/729 is the SG PR that would have saved you some pain

**Comment by drammock** [comment]
> I think this one is ready for review.

reminder to change PR out of "draft" mode, and remove `WIP` from the PR title when it's ready for review.

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -141,6 +141,8 @@ Changelog
 
 - :class:`mne.Report` now can add topomaps of SSP projectors to the generated report. This behavior can be toggled via the new ``projs`` argument by `Richard Höchenberger`_
 
+- Improved documentation building instructions and execution on Windows by `Eric Larson`_ and `kalenkovich`_
```
[the remark this answers is not in the document]
Comment (kalenkovich) [reply doc/changes/latest.inc:144]: It is in `doc/changes/names.inc`

File: doc/install/contributing.rst [hunk 2]
```diff
@@ -231,6 +231,15 @@ MNE-Python, but are needed for locally running our test suite::
...
     $ pip install -r requirements_doc.txt
+    $ conda install graphviz
+
+To build documentation, you will also require `optipng`_:
+
+- On Linux, use the command ``sudo apt install optipng``.
+
+- On MacOS, optipng can be installed using Homebrew.
+
+- On Windows, unzip optipng.exe from the `optipng for Windows`_ archive into the ``doc`` folder.
```
[the remark this answers is not in the document]
Comment (kalenkovich) [reply doc/install/contributing.rst:242]: I thought about this but wanted to keep it consistent with [Configuring git](https://mne.tools/dev/install/contributing.html#id5) in `doc/install/contributing.rst`.

File: doc/changes/latest.inc (hunk 1 above)
[the remark this answers is not in the document]
Comment (kalenkovich) [reply doc/changes/latest.inc:144]: If I want my full name (the handle is my surname), should I change it in `doc/changes/names.inc` and in the previous `something.inc`?

**Comment by kalenkovich** [comment]
> > I think this one is ready for review.
> 
> reminder to change PR out of "draft" mode, and remove `WIP` from the PR title when it's ready for review.

Will do, thank. Waiting for the CIs to pass.

File: doc/changes/latest.inc (hunk 1 above)
[the remark this answers is not in the document]
Comment (drammock) [reply doc/changes/latest.inc:144]: ah, my mistake, I didn't realize you had made other contributions before (I didn't see edits to `names.inc` here, and made a wrong assumption).  I checked and your email is already in our `.mailmap` file so nothing additional needed.

**Comment by kalenkovich** [comment]
> FYI [sphinx-gallery/sphinx-gallery#729](https://github.com/sphinx-gallery/sphinx-gallery/pull/729) is the SG PR that would have saved you some pain

Oh, cool! So, once this PR gets into a new release of sphinx-gallery, an error will be raised when the downloading function asks for a license.

**Comment by kalenkovich** [comment]
> > Should I change this one too?
> 
> It depends on what shell you're using in Windows (command prompt versus mingw or linux subtools, etc.). But feel free to add a note for windows users about this, since probably enough will use the builtin conda prompt.

Yeah, I didn't think about different shells. I think that most of the people who use something beyond cmd/powershell will have no problem figuring out that they don't actually need to use `set` so I did add a note (two, actually).

**Comment by larsoner** [comment]
Thanks @kalenkovich

**Pull request merged**